File: java/src/decorator/io/InputTest.java
package decorator.io;

import java.io.*;
import java.nio.file.Files;
import java.nio.file.Paths;

public class InputTest {
	public static void main(String[] args) throws IOException {
		int c;

		try {
			InputStream in = 
				new LowerCaseInputStream(
					new BufferedInputStream(
                            Files.newInputStream(Paths.get("files/test.txt"))));

			while((c = in.read()) >= 0) {
				System.out.print((char)c);
			}

			in.close();
		} catch (IOException e) {
			e.printStackTrace();
		}
	}
}

File: src/decorator/io/InputTest.kt
package decorator.io

import java.io.BufferedInputStream
import java.io.FileInputStream
import java.io.IOException
import java.io.InputStream


fun main(args: Array<String>) {
    var c: Int

    try {
        val `in`: InputStream =
            LowerCaseInputStream(
                BufferedInputStream(
                    FileInputStream("files/test.txt")
                )
            )

        while ((`in`.read().also{c = it }) >= 0) {
       // while ((c = `in`.read())>=0)
            print(c.toChar())
        }

        `in`.close()
    } catch (e: IOException) {
        e.printStackTrace()
    }
}

